## CI Failure Analysis — commit 63a82b9

**Total failures**: 1 (1 infrastructure)

---

<details>
<summary>🔧 Infrastructure: <code>pull-cluster-network-addons-operator-unit-test</code> — Network failure downloading ubi8/ubi-minimal image blob from cdn01.quay.io during container build - unexpected EOF</summary>

### Error
````
```
Error: creating build container: unable to copy from source docker://registry.access.redhat.com/ubi8/ubi-minimal:latest: copying system image from manifest list: writing blob: storing blob to file "/var/tmp/container_images_storage2629691990/1": happened during read: unexpected EOF (while reconnecting: Get "https://cdn01.quay.io/quayio-production-s3/sha256/28/2879d533a3ed260f6964c746c5cdcb795f4b9f34ebd6c79807cc04faf5045789?...": EOF)
+ rm -rf /tmp/deploy.linux-bridge.fLTg
make[1]: *** [Makefile:246: bump-linux-bridge] Error 125
make: *** [Makefile:248: bump-all] Error 2
```
````

### Root Cause
The build failed while pulling registry.access.redhat.com/ubi8/ubi-minimal:latest during the bump-linux-bridge make target. The failure occurred when downloading a blob from the CDN (cdn01.quay.io S3 storage) with an "unexpected EOF" error during reconnection, indicating a transient network or CDN issue.

### Recommendation
Retest with /retest - this is a transient CDN/network failure when downloading container image blobs from quay.io's S3 storage. The issue should resolve on retry.

</details>

<!-- oompa-bot ci:63a82b9:pull-cluster-network-addons-operator-unit-test -->

--- a/build/operator/Dockerfile
+++ b/build/operator/Dockerfile
@@ -1,13 +1,6 @@
 ARG BUILD_ARCH=amd64
-FROM --platform=linux/${BUILD_ARCH} quay.io/centos/centos:stream9 AS builder
-
-RUN dnf install -y tar gzip jq && dnf clean all
-RUN ARCH=$(uname -m | sed 's/x86_64/amd64/') && \
-    GO_VERSION=$(curl -L -s "https://go.dev/dl/?mode=json" | jq -r '.[0].version') && \
-    curl -L "https://go.dev/dl/${GO_VERSION}.linux-${ARCH}.tar.gz" -o go.tar.gz && \
-    tar -C /usr/local -xzf go.tar.gz && \
-    rm go.tar.gz
-ENV PATH=$PATH:/usr/local/go/bin
+ARG GO_VERSION
+FROM --platform=linux/${BUILD_ARCH} quay.io/projectquay/golang:${GO_VERSION} AS builder
 WORKDIR /go/src/cluster-network-addons-operator
 COPY . .
 

--- a/hack/build-operator-docker.sh
+++ b/hack/build-operator-docker.sh
@@ -7,7 +7,9 @@ fi
 
 IFS=',' read -r -a PLATFORM_LIST <<< "$PLATFORMS"
 
-BUILD_ARGS="--build-arg BUILD_ARCH=$ARCH -f build/operator/Dockerfile -t $OPERATOR_IMAGE_TAGGED --push ."
+GO_VERSION=$(./hack/go-version.sh | cut -d'.' -f1-2)
+
+BUILD_ARGS="--build-arg BUILD_ARCH=$ARCH --build-arg GO_VERSION=$GO_VERSION -f build/operator/Dockerfile -t $OPERATOR_IMAGE_TAGGED --push ."
 
 if [ ${#PLATFORM_LIST[@]} -eq 1 ]; then
     docker build --platform "$PLATFORMS" $BUILD_ARGS

--- a/hack/build-operator-podman.sh
+++ b/hack/build-operator-podman.sh
@@ -7,6 +7,8 @@ fi
 
 IFS=',' read -r -a PLATFORM_LIST <<< "$PLATFORMS"
 
+GO_VERSION=$(./hack/go-version.sh | cut -d'.' -f1-2)
+
 # Remove any existing manifest and image
 podman manifest rm "${OPERATOR_IMAGE_TAGGED}" 2>/dev/null || true
 podman rmi "${OPERATOR_IMAGE_TAGGED}" 2>/dev/null || true
@@ -17,6 +19,7 @@ podman manifest create "${OPERATOR_IMAGE_TAGGED}"
 for platform in "${PLATFORM_LIST[@]}"; do
     podman build \
         --build-arg BUILD_ARCH="$ARCH" \
+        --build-arg GO_VERSION="$GO_VERSION" \
         --platform "$platform" \
         --manifest "${OPERATOR_IMAGE_TAGGED}" \
         -f build/operator/Dockerfile \

